File: lib/app/data/models/date_request_model.dart
class DateRequestModel {
  final int? id;
  final String date;
  final String time;
  final String venue;
  final String? status;
  final int senderId;
  final String? senderName;
  final int receiverId;
  final String? receiverName;
  final bool? sentByReceiver;

  DateRequestModel({
    this.id,
    required this.date,
    required this.time,
    required this.venue,
    this.status,
    required this.senderId,
    this.senderName,
    required this.receiverId,
    this.receiverName,
    this.sentByReceiver,
  });

  factory DateRequestModel.fromJson(Map<String, dynamic> json) {
    return DateRequestModel(
      id: json['id'],
      date: json['date'],
      time: json['time'],
      venue: json['venue'],
      status: json['status'],
      senderId: json['senderId'],
      senderName: json['senderName'],
      receiverId: json['receiverId'],
      receiverName: json['receiverName'],
      sentByReceiver: json['sentByReceiver'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'date': date,
      'time': time,
      'venue': venue,
    };
  }
}

File: server/src/main/java/com/backendProject/SoulSync/date/model/DateRequestModel.java
package com.backendProject.SoulSync.date.model;

import com.backendProject.SoulSync.enums.DateRequestStatus;
import com.backendProject.SoulSync.user.model.UserModel;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Entity
public class DateRequestModel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "sender_id", nullable = false)
    private UserModel sender;

    @ManyToOne
    @JoinColumn(name = "receiver_id", nullable = false)
    private UserModel receiver;

    @NotNull private LocalDate date;
    @NotNull private LocalTime time;
    @NotBlank private String venue;

    @Enumerated(EnumType.STRING)
    private DateRequestStatus status; // PENDING, ACCEPTED, REJECTED

    private boolean sentByReceiver; // tracks who last edited the request

    // Timestamps (optional but helpful)
    @CreationTimestamp
    private LocalDateTime createdAt ;
    @UpdateTimestamp
    private LocalDateTime updatedAt;



    // Getters and Setters

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public UserModel getSender() {
        return sender;
    }

    public void setSender(UserModel sender) {
        this.sender = sender;
    }

    public UserModel getReceiver() {
        return receiver;
    }

    public void setReceiver(UserModel receiver) {
        this.receiver = receiver;
    }

    public LocalDate getDate() {
        return date;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    public LocalTime getTime() {
        return time;
    }

    public void setTime(LocalTime time) {
        this.time = time;
    }

    public String getVenue() {
        return venue;
    }

    public void setVenue(String venue) {
        this.venue = venue;
    }

    public DateRequestStatus getStatus() {
        return status;
    }

    public void setStatus(DateRequestStatus status) {
        this.status = status;
    }

    public boolean isSentByReceiver() {
        return sentByReceiver;
    }

    public void setSentByReceiver(boolean sentByReceiver) {
        this.sentByReceiver = sentByReceiver;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }
}
